app/services/record_declarations/ecf.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module RecordDeclarations
+  class ECF < Base
+    delegate :participant?, to: :user, allow_nil: true
+
+    def schema_validation_params
+      super.merge({ schema_path: "ecf/participant_declarations" })
+    end
+
+    def declaration_type
+      ParticipantDeclaration::ECF
+    end
+
+    def actual_lead_provider
+      SchoolCohort.find_by(school: school, cohort: cohort)&.lead_provider&.cpd_lead_provider
+    end
```

Comment:
`user_profile.school_cohort.lead_provider&.cpd_lead_provider` should work

Reply:
Cool. In that case the need for delegation mentioned before disappears.